File: app/src/main/java/com/pleon/buyt/model/Statistics.java
package com.pleon.buyt.model;

import java.text.NumberFormat;
import java.util.List;

public class Statistics {

    private List<DailyCost> dailyCosts;
    private long totalPurchaseCost;
    private long averagePurchaseCost;
    private Category mostPurchasedCategory;
    private int numberOfPurchases;
    private long maxPurchaseCost;
    private long minPurchaseCost;
    private int weekdayWithMaxPurchases;
    private Store storeWithMaxPurchaseCount;

    public List<DailyCost> getDailyCosts() {
        return dailyCosts;
    }

    public void setDailyCosts(List<DailyCost> dailyCosts) {
        this.dailyCosts = dailyCosts;
    }

    public String getTotalPurchaseCost() {
        return NumberFormat.getInstance().format(totalPurchaseCost);
    }

    public void setTotalPurchaseCost(long totalPurchaseCost) {
        this.totalPurchaseCost = totalPurchaseCost;
    }

    public String getAveragePurchaseCost() {
        return NumberFormat.getInstance().format(averagePurchaseCost);
    }

    public void setAveragePurchaseCost(long averagePurchaseCost) {
        this.averagePurchaseCost = averagePurchaseCost;
    }

    public int getMostPurchasedCategoryName() {
        return mostPurchasedCategory.getNameRes();
    }

    public void setMostPurchasedCategory(Category mostPurchasedCategory) {
        this.mostPurchasedCategory = mostPurchasedCategory;
    }

    public String getNumberOfPurchases() {
        return NumberFormat.getInstance().format(numberOfPurchases);
    }

    public void setNumberOfPurchases(int numberOfPurchases) {
        this.numberOfPurchases = numberOfPurchases;
    }

    public String getMaxPurchaseCost() {
        return NumberFormat.getInstance().format(maxPurchaseCost);
    }

    public void setMaxPurchaseCost(long maxPurchaseCost) {
        this.maxPurchaseCost = maxPurchaseCost;
    }

    public String getMinPurchaseCost() {
        return NumberFormat.getInstance().format(minPurchaseCost);
    }

    public void setMinPurchaseCost(long minPurchaseCost) {
        this.minPurchaseCost = minPurchaseCost;
    }

    public int getWeekdayNameResWithMaxPurchases() {
        return DailyCost.Days.values()[weekdayWithMaxPurchases].getNameStringRes();
    }

    public void setWeekdayWithMaxPurchases(int weekdayWithMaxPurchases) {
        this.weekdayWithMaxPurchases = weekdayWithMaxPurchases;
    }

    public String getStoreNameWithMaxPurchaseCount() {
        return storeWithMaxPurchaseCount.getName();
    }

    public void setStoreWithMaxPurchaseCount(Store storeWithMaxPurchases) {
        this.storeWithMaxPurchaseCount = storeWithMaxPurchases;
    }
}

File: app/src/main/kotlin/com/pleon/buyt/model/Statistics.kt
package com.pleon.buyt.model

import com.pleon.buyt.database.DailyCost
import com.pleon.buyt.database.PieSlice
import java.text.NumberFormat

class Statistics {

    var dailyCosts: List<DailyCost>? = null
    var totalPurchaseCost: Long = 0
    var averagePurchaseCost: Long = 0
    var numberOfPurchases: Int = 0
    var maxPurchaseCost: Long = 0
    var minPurchaseCost: Long = 0
    lateinit var mostPurchasedCategories: List<PieSlice>
    private var weekdayWithMaxPurchases: Int = 0
    private var storeWithMaxPurchaseCount: Store? = null

    val weekdayNameResWithMaxPurchases: Int
        get() {
            return if (totalPurchaseCost != 0L) DailyCost.Days.values()[weekdayWithMaxPurchases].nameStringRes
            else 0
        }

    val storeNameWithMaxPurchaseCount: String
        get() = if (storeWithMaxPurchaseCount == null) "-" else storeWithMaxPurchaseCount!!.name

    fun getTotalPurchaseCost(): String {
        return NumberFormat.getInstance().format(totalPurchaseCost)
    }

    fun getAveragePurchaseCost(): String {
        return NumberFormat.getInstance().format(averagePurchaseCost)
    }

    fun getNumberOfPurchases(): String {
        return NumberFormat.getInstance().format(numberOfPurchases.toLong())
    }

    fun getMaxPurchaseCost(): String {
        return NumberFormat.getInstance().format(maxPurchaseCost)
    }

    fun getMinPurchaseCost(): String {
        return NumberFormat.getInstance().format(minPurchaseCost)
    }

    fun setWeekdayWithMaxPurchases(weekdayWithMaxPurchases: Int) {
        this.weekdayWithMaxPurchases = weekdayWithMaxPurchases
    }

    fun setStoreWithMaxPurchaseCount(storeWithMaxPurchases: Store?) {
        this.storeWithMaxPurchaseCount = storeWithMaxPurchases
    }
}
